.github/workflows/CICD.yml

```diff
@@ -0,0 +1,294 @@
+name: CICD
+
+env:
+  PROJECT_NAME: bat
+  PROJECT_DESC: "A `cat` clone with wings"
+  PROJECT_AUTH: "sharkdp"
+  RUST_MIN_SRV: "1.40.0"
+
+on: [push, pull_request]
+
+jobs:
+  min_version:
+    name: MinSRV # Minimum supported rust version
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install `rust` toolchain (v${{ env.RUST_MIN_SRV }})
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.RUST_MIN_SRV }}
+        default: true
+        profile: minimal # minimal component installation (ie, no documentation)
+    - name: Test
+      uses: actions-rs/cargo@v1
+      with:
+        command: test
+
+  build:
+    name: Build
+    runs-on: ${{ matrix.job.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        job:
+          # { os, target, cargo-options, features, use-cross, toolchain }
+          - { os: ubuntu-latest  , target: arm-unknown-linux-gnueabihf , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: aarch64-unknown-linux-gnu   , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: i586-unknown-linux-gnu      , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: i586-unknown-linux-musl      , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: i686-unknown-linux-gnu      , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: i686-unknown-linux-musl     , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: x86_64-unknown-linux-gnu    , use-cross: use-cross }
+          - { os: ubuntu-18.04   , target: x86_64-unknown-linux-musl   , use-cross: use-cross }
+          - { os: ubuntu-16.04   , target: x86_64-unknown-linux-gnu    , use-cross: use-cross }
+          - { os: macos-latest   , target: x86_64-apple-darwin         }
+          # - { os: windows-latest , target: i686-pc-windows-gnu         }  ## disabled; linker errors
+          - { os: windows-latest , target: i686-pc-windows-msvc        }
+          # - { os: windows-latest , target: x86_64-pc-windows-gnu       }  ## disabled; linker errors ## note: requires rust >= 1.43.0 to link correctly
```

Comment:
1.43 was released month ago.

Reply:
Yes, that's a comment for `coreutils` and a few other projects that I contributed the CICD code too ... and they are all compiling and working. But, here, I see `undefined reference to `_imp____acrt_iob_func'`, for which I don't see a solution yet. I included some references to discussions/issues (some of which you are a part of) about the problem.

- ref: <rust-lang/rust#68887 (comment)>
- ref: <https://users.rust-lang.org/t/linking-with-gcc-failed-exit-code-1/34124>
- ref: <https://www.gitmemory.com/issue/rust-lang/rust/47048/530376978>
- ref: <https://sourceforge.net/p/mingw-w64/mailman/mingw-w64-public/?limit=250&page=7>

I'd be happy to re-enable them here, but v1.43 doesn't fix the noted linker error.
I'll change the comment to be more representative of the situation.

Reply:
That's because you don't have i686 MinGW installed (GH Actions provide only x86_64), note that x86_64 build passed and failed tests.
It will work if you use MSYS2 which provides both toolchains and was added to GHA recently https://github.com/actions/virtual-environments/issues/30

Reply:
Thanks for the insight!
I'll attempt a new build using the MSYS toolchain.
Any suggestions on a process? I haven't ever tried using that toolchain on GHA.

Reply:
We had MSVC + GNU builds some time ago (in `bat` or `fd`?) and someone suggested that we only ship one of them in order not to confuse users. I'd be okay with only providing the MSVC version.